pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -94,13 +138,18 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 					// add to cache if the pod is running or pending
 					pc.update(ip, key)
 				}
+				if _, ok := pc.podTransitionTime[ip]; !ok && IsPodReady(pod) {
```

Comment:
using IP is not a reliable key, pods in host network may share an IP

Reply:
Also terminated pods can

Reply:
Correct. But is the existing logic also broken for host network?

Reply:
@hzxuzhonghu just had a PR to fix this I think? It was just merged, maybe you need to rebase?